lib/v2/screens/send_scanner/send_scanner_screen.dart

```diff
@@ -0,0 +1,94 @@
+  final _sendPageBloc = SendPageBloc();
+
+  @override
+  void initState() {
+    super.initState();
+    _scannerScreen = ScannerScreen(resultCallback: onResult);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(title: const Text("Scan QR Code")),
+      body: BlocProvider(
+        create: (context) => _sendPageBloc,
+        child: Column(
+          children: [
+            const SizedBox(height: 32),
+            Text("Scan QR Code to Send", style: Theme.of(context).textTheme.button),
+            const SizedBox(height: 32),
+            _scannerScreen,
+            BlocBuilder<SendPageBloc, SendPageState>(
+              builder: (context, SendPageState state) {
+                switch (state.pageState) {
+                  case PageState.initial:
+                    _scannerScreen.scan();
+                    return const SizedBox.shrink();
+                  case PageState.loading:
+                    _scannerScreen.showLoading();
+                    return const SizedBox.shrink();
+                  case PageState.failure:
+                    return Padding(
+                      padding: const EdgeInsets.all(32),
+                      child: Container(
+                          padding: const EdgeInsets.all(16),
+                          width: double.infinity,
+                          child: Padding(
+                            padding: const EdgeInsets.all(8.0),
+                            child: Text(
+                              state.error,
+                              style: Theme.of(context).textTheme.subtitle2.copyWith(color: AppColors.orangeYellow),
+                              textAlign: TextAlign.center,
+                            ),
+                          ),
+                          decoration: const BoxDecoration(
+                              color: AppColors.black, borderRadius: BorderRadius.all(Radius.circular(8)))),
+                    );
+                  case PageState.success:
+                    _scannerScreen.stop();
+                    NavigationService.of(context).navigateTo(
+                        Routes.customTransaction,
+                        CustomTransactionArguments(
+                          account: state.pageCommand.resultData.accountName,
+                          name: state.pageCommand.resultData.name,
+                          data: state.pageCommand.resultData.data,
+                        ),
+                        true);
+                    return Text(
+                      "Success",
+                      style: Theme.of(context).textTheme.caption,
+                    );
+                  default:
+                    return const SizedBox.shrink();
+                }
+              },
+            )
+          ],
+        ),
+      ),
+    );
+  }
+
+  Future<void> onResult(String scanResult) async {
+    _sendPageBloc.add(ExecuteScanResult(scanResult: scanResult));
```

Comment:
is there a better way to handle the bloc? I saved it in a private var but i dont like it.

Reply:
If you mean the `_sendPageBloc` I don't think it's possible since you need a reference to it to trigger the event. 
And since you only need to instantiate `_scannerScreen` once (i.e. in the initState) you need to create the bloc as you have it (`final _sendPageBloc = SendPageBloc();`) or in the initState too.


Another approach would be to use a `BlocListener` under `BlocProvider` so you don't create a reference to the bloc you can access by:
```
// with extensions
context.read<BlocA>();

// without extensions
BlocProvider.of<BlocA>(context)
```
But the problem is you need the block.add inside the onResult method that is passed as a callback